controller_manager:
  ros__parameters:
    update_rate: 100 # Hz
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    
    diff_drive_controller:
      type: diff_drive_controller/DiffDriveController
    
    imu_sensor_broadcaster:
      type: imu_sensor_broadcaster/IMUSensorBroadcaster


diff_drive_controller:
  ros__parameters:
    left_wheel_names: ["Revolute_Left_Wheel"]
    right_wheel_names: ["Revolute_Right_Wheel"]
    wheel_separation: 0.173
    wheel_radius: 0.095
    right_wheel_radius_multiplier: -1.0  # give opposite direction commands to right motor
    base_frame_id: "base_link"
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    position_feedback: false
    linear.x.max_velocity: 10.0 # Max forward velocity in m/s
  
imu_sensor_broadcaster:
  ros__parameters:
    sensor_name: "BNO08X"
    frame_id: "imu"